onnxruntime/core/providers/webgpu/tensor/pad.cc

```diff
@@ -28,74 +29,58 @@ Status PadProgram<T>::GenerateShaderCode(ShaderHelper& shader) const {
                             << "  var use_pad_value = false;\n"
                             << "  var in_coord = i32(0);\n";
 
-  std::string shapeDimStr = output.Rank() == 1 ? "" : "[dim]";
-  std::string strideDimStr = output.Rank() < 3 ? "" : "[dim]";
-  std::string begin_axis_statement, end_axis_statement;
-  std::string in_axis_statement = "in_coord = i32(output_indices" + shapeDimStr + ") - uniforms.lower_pads" +
-                                  shapeDimStr + ";\n";
+  const int rank = output.Rank();
+  std::string output_indices_str = rank == 1 ? "i32(output_indices)" : "i32(output_indices[dim])";
+  std::string lower_pads_str = rank == 1 ? "uniforms.lower_pads" : "uniforms.lower_pads[dim]";
```

Comment:
you can use `GetElementAt()` or `ShaderIndicesHelper::IndicesGet()` to generate those strings. The implementation of these 2 functions are basically doing the same thing you wrote

Reply:
Done